src/ipcache.cc

```diff
@@ -320,8 +320,13 @@ ipcacheRelease(ipcache_entry * i, bool dofree)
 static ipcache_entry *
 ipcache_get(const char *name)
 {
-    if (ip_table != nullptr)
-        return (ipcache_entry *) hash_lookup(ip_table, name);
+    if (ip_table) {
+        char *newName = xstrdup(name);
+        Tolower(newName);
+        ipcache_entry *entry = (ipcache_entry *) hash_lookup(ip_table, newName);
+        free(newName);
+        return entry;
```

Comment:
> At that replacement time, the developer should investigate whether our ClpMap should be enhanced to be reusable in this context

I had investigated that. It is currently not useable as it is not currently possible to iterate over the contents of a ClpMap (e.g. for displaying the contents in Cache Manager). Extending ClpMap is in my sights

Reply:
> > Alex: At that replacement time, the developer should investigate whether our ClpMap should be enhanced to be reusable in this context

> Francesco: I had investigated that. It is currently not useable

That is why I said "investigate whether our ClpMap should be _enhanced_ to be reusable" rather than simply "reuse our ClpMap". It sounds like we agree that ClpMap is not reusable "as is" in this context. It is not yet known whether it should be enhanced to become reusable, but I bet that investigation will yield a positive answer.

However, that is all for the long term (unless somebody volunteers to do the associated heavy lifting right now). For now, IMO, we can just fix the (already configurable) hashing and comparison functions of the existing `ip_table` and continue to use the legacy `hash_table` type.

Reply:
@rousskov everyone Thanks for looking at it. @rousskov its not just perf fix as with squid 5.9. with mix case domain names, intercept mode is not working at all.

@rousskov I had a look at code. It seems that providing case insensitive hash function should be easy. I did not understand case insensitive hash function. We are currently using strcmp function as comparison function for iptable here https://github.com/squid-cache/squid/blob/master/src/ipcache.cc#L705. If we convert this to strcasecmp we should be good ?

Reply:
I found that we are already using strcasecmp at multiple places in code and its mostly implemented everywhere. So used it.

Can you please review @rousskov @kinkie @yadij

Reply:
> its not just perf fix as with squid 5.9. with mix case domain names, intercept mode is not working at all.

I am aware of your observation, but the problems you have observed have negligible effect on the classification: ipcache_get() is a cache lookup function. _Correct_ code will continue to work correctly even if we replace the entire ipcache_get() code with "return nullptr". It will be slower, but it will continue to work. Yes, some buggy Squid code and some Squid misconfigurations will stop working if this cache is disabled, but, as far as the primary classification is concerned, those are very _different_ problems compared to false ipcache_get() misses that these PR code changes are fixing.

I am not against mentioning your observations in the PR description (i.e. the future official commit message). We just need to stay vague enough because we do not know much about the true nature of the problems you have observed.

> If we convert this to strcasecmp we should be good ?

Since this change request got outdated, I moved this part of the discussion to a fresh change request dedicated to the hash function bug.

I consider _this_ change request addressed. Thank you!